### PR TITLE
Fix double encoding for dynamic field filters in ticket overviews

### PR DESCRIPTION
**Pull request by tipue-dev**
## Proposed change

Dynamic field keys as filter values are encoded twice. Therefore it is not possible to filter for tree components from select fields (e.g. Main::Sub).

This was a bug before (bug#[14497](https://bugs.otrs.org/show_bug.cgi?id=14497)) and was fixed in ce5980f183d4c363ad03571629478d94c6a71b5b. A second fix for another bug (bug#[14982](https://bugs.otrs.org/show_bug.cgi?id=14982)) (not encoded CustomerID) in ecfeb7aa300eac654f9d0eafaa2d3b80214607d5 broke the first fix by encoding the key in the frontend, while the first fix already encoded the dynamic field values in the backend (e.g. "Main::Sub" -> "Main%3A%3ASub" -> "Main%253A%253ASub").

Since the later fix also works on the first bug, the first fix can be removed.

## Type of change

-  🆙 - Dependency upgrade (e.g. libraries) 
- [x] 🐞 - Bugfix (non-breaking change which fixes an issue) 
- [x] 💎 - Code quality improvements to existing code or addition of unit tests 
-  🚀 - New feature (which adds functionality to an existing integration) 

## Additional information

To replicate the error:
 - Create a dynamic field (type dropdown) and create the keys "Main" and "Main::Sub".
 - Create a Ticket and add the dynamic field value with the key "Main::Sub".
 - Add the dynamic field to the setting `DashboardBackend###0120-TicketNew`.
 - Go to the dashboard widget "New Tickets" and select the matching value for the key "Main::Sub"
Expected result: the created ticket is shown.
Current result: no ticket is shown.

The same is true for the actions using TicketOverview/Small.pm, like AgentTicketQueue and AgentTicketStatusView.

## Checklist

- [x] The code change is tested and works locally.(❗)
- [x] There is no commented out code in this PR.(❕)
- [x] You improved or added new unit tests.(❕)
- [x] Local ZnunyCodePolicy run passes successfully.(❕)
- [x] Local unit tests pass.(❕)
- [x] GitHub workflow ZnunyCodePolicy passes.(❗)
- [x] GitHub workflow unit tests pass.(❗)

### CODE REVIEW

File: Kernel/Output/HTML/Dashboard/TicketGeneric.pm [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # --
 # Copyright (C) 2001-2021 OTRS AG, https://otrs.com/
 # Copyright (C) 2021 Znuny GmbH, https://znuny.org/
+# Copyright (C) 2021 maxence business consulting GmbH, http://www.maxence.de
```
Comment (dennykorsukewitz) [inline Kernel/Output/HTML/Dashboard/TicketGeneric.pm]:
```suggestion
```
We have changed our rules regarding copyright.
We therefore ask you to remove your newly added copyrights.

```
Znuny is an open fork of an existing software. 
So we have to respect the already given copyright of the original creators.

New files will be licensed using the AGPL Version 3. 
If you contribute code to the Znuny project you will get mentioned in the pull request 
incl. the commit, in CHANGES.md and in AUTHORS.md. 
We will not mention you in the file you provided or changed. 
Your work is highly appreciated and acknowledged 
but you contribute it to the project and your copyright will pass on to the fork itself.
```

https://github.com/znuny/Znuny/blob/rel-6_1-dev/CONTRIBUTING.md#licensing-copyright-and-credits

File: Kernel/Output/HTML/TicketOverview/Small.pm [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # --
 # Copyright (C) 2001-2021 OTRS AG, https://otrs.com/
 # Copyright (C) 2021 Znuny GmbH, https://znuny.org/
+# Copyright (C) 2021 maxence business consulting GmbH, http://www.maxence.de
```
Comment (dennykorsukewitz) [inline Kernel/Output/HTML/TicketOverview/Small.pm]:
```suggestion
```

File: scripts/test/Selenium/Output/Dashboard/TicketGenericFilter.t [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # --
 # Copyright (C) 2001-2021 OTRS AG, https://otrs.com/
 # Copyright (C) 2021 Znuny GmbH, https://znuny.org/
+# Copyright (C) 2021 maxence business consulting GmbH, http://www.maxence.de
```
Comment (dennykorsukewitz) [inline scripts/test/Selenium/Output/Dashboard/TicketGenericFilter.t]:
```suggestion
```

File: scripts/test/Selenium/Output/Ticket/OverviewSmall.t [hunk 4]
```diff
@@ -1,6 +1,7 @@
 # --
 # Copyright (C) 2001-2021 OTRS AG, https://otrs.com/
 # Copyright (C) 2021 Znuny GmbH, https://znuny.org/
+# Copyright (C) 2021 maxence business consulting GmbH, http://www.maxence.de
```
Comment (dennykorsukewitz) [inline scripts/test/Selenium/Output/Ticket/OverviewSmall.t]:
```suggestion
```

**Review by dennykorsukewitz — CHANGES_REQUESTED** [review]
Hi @tipue-dev,

nice PR 🚀

**Comment by tipue-dev** [comment]
> nice PR 🚀

Thank you :blush:

**Review by dennykorsukewitz — APPROVED** [review]

**Review by dennykorsukewitz — APPROVED** [review]

**Pull request merged**